### PR TITLE
Add a check to the DbUnitChecker that interesting CALC PVs should be readonly

### PR DESCRIPTION
**Pull request by samueljackson92**
This adds a check to the `DbUnitChecker` to make sure that any `calc`, `calcout`, and `transform` PVs that are marked as interesting are also have the EPCIS ASG set to readonly.

This is part of issue ISISComputingGroup/IBEX#2343
The main PR for this is [here.](https://github.com/ISISComputingGroup/EPICS-ioc/pull/155)

### CODE REVIEW

File: db_checker.py [hunk 1]
```diff
@@ -92,6 +95,23 @@ def test_interest_units(self):
                         print "ERROR: Missing units on " + err_src_fmt(db, rec)
 
         self.assertEqual(err, 0, msg="Missing units on interesting PVs in project")
+		
+	def test_interest_calc_readonly(self):
```
Comment (Tom-Willemsen) [inline db_checker.py]: Style (PEP8): please don't mix tabs and spaces for indentation. Our standard is to use spaces.

File: db_checker.py [hunk 2]
```diff
@@ -92,6 +95,23 @@ def test_interest_units(self):
...
+		"""
+		This method checks that interesting PVs have units
```
Comment (Tom-Willemsen) [inline db_checker.py]: This is not what this method does. Please change.

File: db_checker.py [hunk 3]
```diff
@@ -92,6 +95,23 @@ def test_interest_units(self):
...
+        for db in dbs:
+            for rec in db.records:
+                if rec.is_interest() and not rec.is_disable() and (rec.get_type() in ASG_list):
+                    value = rec.get_field("ASG")
+
+                    if value != "READONLY":
+                        err += 1
+                        print "ERROR: Missing ASG on " + err_src_fmt(db, rec)
+
+        self.assertEqual(err, 0, msg="Missing ASG on interesting PVs in project")
```
Comment (Tom-Willemsen) [inline db_checker.py]: Can you make this error message a bit more specific, e.g. "Missing ASG on interesting calculation records in project"

File: db_checker.py [hunk 4]
```diff
@@ -92,6 +95,23 @@ def test_interest_units(self):
...
+		
+	def test_interest_calc_readonly(self):
+		"""
+		This method checks that interesting PVs have units
+		"""
+        err = 0
+
+        for db in dbs:
+            for rec in db.records:
+                if rec.is_interest() and not rec.is_disable() and (rec.get_type() in ASG_list):
```
Comment (Tom-Willemsen) [inline db_checker.py]: Why is the `rec.is_disable()` condition here? 

The disable record is always a `bo` record anyway, so it seems irrelevant.

**Review by Tom-Willemsen — CHANGES_REQUESTED** [review]

File: db_checker.py (hunk 4 above)
Comment (samueljackson92) [reply db_checker.py]: This was copied from the test above. Does it also need to be removed from `test_interest_units` as well?
